### PR TITLE
Draft: Add go linter tool and unit testing for Pr build process

### PR DESCRIPTION
**Pull request by n-th**
Signed-off-by: Nathalia <nathalia.gomazako@gmail.com>

**Pull Request check list**

- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
Add go linter and unit testing for quality code assurance. Check more clicking [here](https://golangci-lint.run/).

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

**Review by mchurichi — APPROVED** [review]
🚀

**Review by wibarre — DISMISSED** [review]

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+    - uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923 # v3.2.0
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: Lint
+      uses: golangci/golangci-lint-action@537aa1903e5d359d0b27dbc19ddd22c5087f3fbc # v3.2.0
+      with:
+        go-version: ${{ env.GO_VERSION }}
```
Comment (mchurichi) [inline .github/workflows/linter.yml:19]:
```suggestion
        version: ${{ env.GO_VERSION }}
```

Comment (n-th) [reply .github/workflows/linter.yml]: Thanks, just changed it. version is the version of the golangci that is required and the go-version is automatically detected.

**Comment by n-th** [comment]
Withs this tool, we can have this feedbacks: 

image

**Review by mchurichi — APPROVED** [review]
🌭

**Review by jufantozzi — APPROVED** [review]
🌾

**Pull request merged**